client/src/common/dataProvider.js

```diff
@@ -0,0 +1,76 @@
+import { stringify } from "query-string";
+import axios from "axios";
+
+import config from "../config";
+
+const pkDictionary = {
```

Comment:
I'd like to request a change here, please.  Could this dictionary be passed into the `dataProvider`?  Then the `dataProvider` doesn't need to know about our specific `chinook.db` database.

Also, when you're doing the `id` swap, you would use something like:
```
cont primaryKey = pkDictionary[resource]
if (primaryKey !== undefined) {
  item.id = item[primaryKey]
}
```

That would avoid having to declare the primary keys for a dictionary that uses plain `id` fields.

Reply:
Hello Ian, can you please elaborate this comment, currently the `pkDictionary` is in the `dataProvider.js` file, do you want me to add the `pkDictionary` as an object like this 

```js 
export const dataProvider = {
  getList: (resource, params) => {}, 
  getOne: (resource, params) => {}, 
  configs: {pkDictionary: {} }
}
```

Reply:
Sure, I'd like to remove `dataProvider`s knowledge of the structure of the chinook database.  So, the primary keys would be defined in `app.js`, and passed in as a parameter of `dataProvider`.  I'm pretty sure other react-admin data-providers take parameters, particularly for the URL to use.  Aah, the URL should be passed in too, please.   Then, no changes need to be made to `dataProvider` in order to support a different `soul` database.

```
  export const dataProvider = (pkDictionary) => {
  getList: (resource, params) => {
    const primaryKey = pkDictionary[resource]
}, 
  getOne: (resource, params) => {}, 
  configs: {pkDictionary: {} }
}
```